File: py-algorithm/arrays/find_the_peaks.py
import unittest

from math import gcd
from fractions import Fraction
from typing import List
from queue import PriorityQueue
from collections import Counter
import itertools

"""
    https://leetcode.com/problems/find-the-peaks/
    
    Bạn được cho mảng mountain. Hãy tìm các chỉ số của phần tử là đỉnh, tức là lớn
    hơn cả hai phần tử lân cận.

    Lưu ý: phần tử đầu và cuối không được coi là đỉnh.
"""


class Solution(unittest.TestCase):

    def findPeaks(self, mountain: List[int]) -> List[int]:
        result = []
        for i in range(1, len(mountain) - 1):
            if mountain[i - 1] < mountain[i] and mountain[i] > mountain[i + 1]:
                result.append(i)

        return result

    def test_1(self):
        self.assertEqual([], self.findPeaks([2, 4, 4]))
        self.assertEqual([1, 3], self.findPeaks([1, 4, 3, 8, 5]))


if __name__ == '__main__':
    unittest.main()

File: leetcode/src/main/java/array/FindThePeaks.java
package array;

import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

/*-
    https://leetcode.com/problems/find-the-peaks/

    Bạn được cho mảng mountain. Hãy tìm các chỉ số của phần tử là đỉnh, tức là lớn
    hơn cả hai phần tử lân cận.

    Lưu ý: phần tử đầu và cuối không được coi là đỉnh.
 */
public class FindThePeaks {

    public List<Integer> findPeaks(int[] mountain) {
        ArrayList<Integer> result = new ArrayList<>();
        for (int i = 1; i < mountain.length - 1; i++) {
            if (mountain[i - 1] < mountain[i] && mountain[i] > mountain[i + 1]) {
                result.add(i);
            }
        }
        return result;
    }

    @Test
    public void test() {
        assertIterableEquals(Collections.emptyList(), findPeaks(new int[]{2, 4, 4}));
        assertIterableEquals(Arrays.asList(1, 3), findPeaks(new int[]{1, 4, 3, 8, 5}));
    }

}
